### PR TITLE
101 Add cross-reference support to xml2yaml conversion

### LINKED ISSUE #101: Cross referenced fields in xml is not converted to valid yaml / jinja template in ioc.yaml
## Steps To Reproduce

In an ioc-pmac devcontainer with BL03I-BUILDER release 2026-Run2-1 in /workspaces/

```
$ uv pip install builder2ibek
$ uv run builder2ibek --version
2.0.1
$ uv run builder2ibek xml2yaml /workspaces/BL03I-BUILDER/etc/makeIocs/BL03I-MO-IOC-11.xml --yaml ioc.yaml
```
In the xml, field ERES refers to MRES:
```
(master) [kwd30084@ws456 BL03I-BUILDER]$ cat etc/makeIocs/BL03I-MO-IOC-11.xml | grep dls_pmac_asyn_motor

<pmac.dls_pmac_asyn_motor ACCL="0.1" ADDR="5" BACC="0.5" BVEL="0.2" DESC="QBPM1 Y Translation" 
DHLM="1.1" DLLM="-41" EGU="mm" ERES="$(MRES)" HOME="BL03I-DI-QBPM-01" HOMEVIS="" 
JAR="1" M=":Y" MRES="0.0001" OFF="0" P="BL03I-DI-QBPM-01" PORT="PPMAC01" PREC="4" 
SPORT="PPMAC01_PORT" SREV="200" TWV="1" VELO="2.5" VMAX="3" gda_desc="Pre-HFM QBPM Y" 
gda_name="QBPM1.Y" name="QBPM1.Y"/>
```

The generated `ioc.yaml` contains:

```
    ERES: $(MRES)
    FOFF: Variable
    HLM: 0.0
    HOME: BL03I-DI-QBPM-01
    JAR: 1
    LLM: 0.0

```
where ERES here is an invalid expression. Failing with pydantic pattern errors:
```
    entities.7.`pmac.dls_pmac_asyn_motor`.HLM
      Input should be a valid string [type=string_type, input_value=0.0, input_type=float]
        For further information visit https://errors.pydantic.dev/2.12/v/string_type
    entities.7.`pmac.dls_pmac_asyn_motor`.LLM
      Input should be a valid string [type=string_type, input_value=0.0, input_type=float]
        For further information visit https://errors.pydantic.dev/2.12/v/string_type
    entities.7.`pmac.dls_pmac_asyn_motor`.ERES.constrained-str
      String should match pattern '.*\{\{.*\}\}.*' [type=string_pattern_mismatch, input_value='$(MRES)', input_type=str]
        For further information visit https://errors.pydantic.dev/2.12/v/string_pattern_mismatch
    entities.7.`pmac.dls_pmac_asyn_motor`.ERES.float
      Input should be a valid number, unable to parse string as a number [type=float_parsing, input_value='$(MRES)', input_type=str]
```

### PR DESCRIPTION
**Pull request by kgabor**
XML attributes containing `$(FIELD)` cross-references are now converted to Jinja2 templates with appropriate type filters. The type is inferred from the referenced field's value within the same element. No recursion or any expression evaluation, simple `$(FIELD)` references are recognised only. Fix #101 

## Summary by CodeRabbit

* **New Features**
  * Added XML attribute cross-reference detection and resolution. The converter now recognizes cross-referenced attribute values, automatically determines appropriate data types, and handles edge cases like leading zeros.

* **Tests**
  * Added comprehensive test samples including YAML configurations, XML setup files, and generated EPICS startup scripts to validate cross-reference functionality.

### CODE REVIEW

**Review by gilesknap — COMMENTED** [review]
Nice work. Thanks for this. 

Please could you add one test that demonstrates it?

Cheers!

File: tests/samples/cross_ref_test.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    ADDR: 5
+    BACC: 0.5
+    BVEL: 0.2
+    Controller: PPMAC01
+    DESC: QBPM1 Y Translation
+    DHLM: 1.1
+    DIR: Pos
+    DLLM: -41
+    EGU: mm
+    ERES: '{{ MRES | float }}'
```
Comment (gilesknap) [inline tests/samples/cross_ref_test.yaml:43]: hi @kgabor this tells me that the rendering has not worked.

The resolved value should be seen in the substitution file as the jinja processing should already have occurred.

If you need help debugging this then let me know.

**Review by gilesknap — COMMENTED** [review]
See inline comment. What you did looks right to me but is not working.

Also please don't delete coderabbit comments - just let it do its thing. It will tidy up at some point after you push your changes.

File: tests/samples/cross_ref_test.yaml (hunk 1 above)
Comment (kgabor) [reply tests/samples/cross_ref_test.yaml:43]: OK, I'll look into it.

Comment (kgabor) [reply tests/samples/cross_ref_test.yaml:43]: Sorry, I'm confused here. `bl03i-mo-ioc-11.yaml` is the output of `builder2ibek xml2yaml`, isn't it? That's what we expect here, and it does contain other jinja2 templates like `ioc_name: "{{  _global.get_env('IOC_NAME') }}"`. It's correctly substituted in the ioc.subst file by `ibek runtime generate2` and both MRES and ERES become `0.0001` in this case.

Comment (gilesknap) [reply tests/samples/cross_ref_test.yaml:43]: Sorry. Its me thats confused, not sure what I was doing there. 

Your test works perfectly and the PR is ready to merge.

**Review by gilesknap — APPROVED** [review]

**Comment by gilesknap** [comment]
Thanks @kgabor this is really nice. Merging and releasing
